### DISCUSSION TITLE
Wazuh throwing error " Error Pattern Handler " (getpatternlist)

### DISCUSSION THREAD

**Discussion by naikankit** [wazuh/wazuh#16538] [General]
circuit_breaking_exception
[parent] Data too large, data for [<reduce_aggs>] would be [1026292632/978.7mb], which is larger than the limit of [1020054732/972.7mb], real usage: [1026292632/978.7mb], new bytes reserved: [0/0b], usages [request=0/0b, fielddata=2781329/2.6mb, in_flight_requests=3952/3.8kb, accounting=47510004/45.3mb]

Error: Too Many Requests
    at Fetch._callee3$ (https://wazuh.domain.com/36136/bundles/core/core.entry.js:6:59535)
    at tryCatch (https://wazuh.domain.com/36136/bundles/plugin/opendistroQueryWorkbenchKibana/opendistroQueryWorkbenchKibana.plugin.js:1:32004)
    at Generator.invoke [as _invoke] (https://wazuh.domain.com/36136/bundles/plugin/opendistroQueryWorkbenchKibana/opendistroQueryWorkbenchKibana.plugin.js:1:35968)
    at forEach.prototype.<computed> [as next] (https://wazuh.domain.com/36136/bundles/plugin/opendistroQueryWorkbenchKibana/opendistroQueryWorkbenchKibana.plugin.js:1:33129)
    at fetch_asyncGeneratorStep (https://wazuh.domain.com/36136/bundles/core/core.entry.js:6:52652)
    at _next (https://wazuh.domain.com/36136/bundles/core/core.entry.js:6:52968)

I have replaced actual domain with "domain.com" for security reasons

**Comment by mdiego92** [comment]
Hello @naikankit,

The circuit_breaking_exception is a mechanism used to prevent operations from causing an OutOfMemoryError. It seems like Wazuh Indexer was using most of the JVM heap configured. The total memory required for all operations was superior to the memory available, so your requested operation was aborted.

We suggest raising the JVM configured but without surpassing more than 50% of available RAM in the `/etc/wazuh-indexer/jvm.options` for this issue.

- Article for reference: [Elasticsearch tuning - Elasticsearch · Wazuh documentation](https://documentation.wazuh.com/current/user-manual/elasticsearch/elastic-tuning.html#memory-locking)

If you’re already using 50% of the available RAM it is not recommended to continue increasing it. In this case, we advise:

- Consider scaling the cluster to obtain more JVM memory to support your workload. This means adding more nodes to the Wazuh-indexer/Elasticsearch cluster.
- If cluster scaling isn't possible, try reducing the number of shards by deleting old or unused indices. Because shard metadata is stored in memory, reducing the number of shards can reduce overall memory usage.
- Change the shards configuration for indices: [https://www.elastic.co/es/blog/how-many-shards-should-i-have-in-my-elasticsearch-cluster](https://www.google.com/url?q=https%3A%2F%2Fwww.elastic.co%2Fes%2Fblog%2Fhow-many-shards-should-i-have-in-my-elasticsearch-cluster&sa=D&sntz=1&usg=AOvVaw3J5cebg62xn6I4YKreDrN5)
- Reduce incoming traffic to your cluster, especially if you have a heavy workload.

Please let me know if you have any remaining questions!

**Reply by naikankit** [reply]
Hi @mdiego92 

Thank You, I was able to fine tune, delete some old indexes. Just curious to know I see indexes created for each day. If I delete any old index does that delete actual log files ?
